File: Data Structures & Algorithms/length-of-last-word/submission-3.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        n = len(s) -1
        while n >= 0:
            if s[n] != " ":
                break
            n -= 1

        count =0
        while s[n] != " " and n >= 0:
            count += 1
            n -= 1
        
        return count

File: Data Structures & Algorithms/anagram-groups/submission-3.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> hm = new HashMap<>();

        for(int i =0; i<strs.length; i++) {
            int[] freqArr = new int[26];

            for(char c: strs[i].toCharArray()) {
                freqArr[c - 'a']++;
            }

            StringBuilder key = new StringBuilder();
            for(int count: freqArr) {
                key.append("#").append(count);
            }

            hm.computeIfAbsent(key.toString(),
                k -> new ArrayList<>())
            .add(strs[i]);
        }
        return new ArrayList<>(hm.values());
        
    }
}
